lib/segment/src/index/field_index/numeric_index.rs

```diff
@@ -380,18 +378,17 @@ impl<T: Encodable + Numericable> PayloadFieldIndex for NumericIndex<T> {
         &self,
         condition: &FieldCondition,
     ) -> OperationResult<CardinalityEstimation> {
-        match condition.range.as_ref().map(|range| {
-            let mut cardinality = self.range_cardinality(range);
-            cardinality
-                .primary_clauses
-                .push(PrimaryCondition::Condition(condition.clone()));
-            cardinality
-        }) {
-            Some(cardinality) => Ok(cardinality),
-            None => Err(OperationError::service_error(
-                "failed to estimate cardinality",
-            )),
-        }
+        condition
```

Comment:
cool! That looks much nicer than `match` I had originally 👍.  Will consider this approach next time.